core-kotlin-modules/core-kotlin-9/src/test/kotlin/com/baeldung/ifvslet/IfVsLetUnitTest.kt

```diff
@@ -0,0 +1,35 @@
+package com.baeldung.ifvslet
+
+import org.junit.jupiter.api.Test
+import kotlin.random.Random
+import kotlin.test.assertEquals
+import kotlin.test.assertNull
+
+class IfVsLetUnitTest {
+
+    @Test
+    fun `Using let to uppercase string`() {
+        val name = fetchNameFromDatabase()
+        val uppercased = name?.let { it.uppercase() }
+
+        if (name == null) {
+            assertNull(uppercased)
+        } else {
+            assertEquals("NAME", uppercased)
+        }
+    }
+
+    @Test
+    fun `Using if x not null to uppercase string`() {
+        val name = fetchNameFromDatabase()
+        val uppercased = if(name != null) name.uppercase() else null
+
+        if (name == null) {
+            assertNull(uppercased)
+        } else {
+            assertEquals("NAME", uppercased)
+        }
+    }
+
+    fun fetchNameFromDatabase(): String? = if(Random.nextBoolean()) "name" else null
```

Comment:
Let's re-work this to not have randomized tests. Can we just split this into 4 tests - one each for null and non-null?